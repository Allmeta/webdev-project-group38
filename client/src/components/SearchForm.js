import React from 'react'
import { connect } from 'react-redux'
import { Form, Icon, Header } from 'semantic-ui-react'
import {fetchMovies, updateGenre, updateTitle} from "../actions/SearchFormActions";

function SearchForm(props) {
  return (
    <Form loading={props.loading} onSubmit={props.handleSubmit}>
      <Header>Search</Header>
      <Form.Group>
        <Form.Input onChange={props.handleChange} value={props.title} name='title' placeholder='Title' width={3}/>
        <Form.Input onChange={props.handleChange} value={props.genre} name='genre' placeholder='Genre' width={3}/>
        <Form.Button type='submit' width={2} icon={<Icon name='search'/>}/>
      </Form.Group>
    </Form>
  )
}

function mapStateToProps(state){
  return({
      title: state.title,
      genre: state.genre,
      loading: state.loading
    }
  )
}

function mapDispatchToProps(dispatch) {
  return (
    {
      handleChange: (e, {name, value}) => {
        if (name === 'title') {
          dispatch(updateTitle(value));
        }
        else if( name === 'genre' ){
          dispatch(updateGenre(value));
        }
      },
      handleSubmit: () => {
        dispatch(fetchMovies());
      }
    }
  )
}

export default connect(mapStateToProps, mapDispatchToProps)(SearchForm);