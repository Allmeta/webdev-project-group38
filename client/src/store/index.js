import { createStore, applyMiddleware, compose } from 'redux'
import { SearchFormReducer } from "../reducers/SearchFormReducer";
import thunk from 'redux-thunk'

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(SearchFormReducer, composeEnhancers(
  applyMiddleware(thunk)
));

export default store;