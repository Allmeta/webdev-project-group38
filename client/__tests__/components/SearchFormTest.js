import React from 'react'
import SearchForm from '../../src/components/SearchForm.js'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import store from '../../src/store/index.js'
import { Provider } from 'react-redux'

it('renders correctly', () => {
  const tree = renderer.create(<Provider store={store}><SearchForm /></Provider>).toJSON()
  expect(tree).toMatchSnapshot()
})
