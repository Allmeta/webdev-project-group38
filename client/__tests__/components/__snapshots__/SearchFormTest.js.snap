// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className="ui form"
  onSubmit={[Function]}
>
  <div
    className="field"
  >
    <div
      className="ui action input"
    >
      <input
        name="title"
        onChange={[Function]}
        placeholder="Search"
        type="text"
      />
      <button
        className="ui icon button"
        onClick={[Function]}
        role="button"
        type="submit"
      >
        <i
          aria-hidden="true"
          className="search icon"
        />
      </button>
    </div>
  </div>
</form>
`;
